src/main/java/com/minecolonies/network/messages/ColonyViewWorkOrderMessage.java

```diff
@@ -0,0 +1,98 @@
+package com.minecolonies.network.messages;
+
+import com.minecolonies.colony.CitizenData;
+import com.minecolonies.colony.Colony;
+import com.minecolonies.colony.ColonyManager;
+import com.minecolonies.colony.WorkOrderView;
+import com.minecolonies.colony.workorders.AbstractWorkOrder;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessage;
+import net.minecraftforge.fml.common.network.simpleimpl.IMessageHandler;
+import net.minecraftforge.fml.common.network.simpleimpl.MessageContext;
+
+/**
+ * Add or Update a ColonyView on the client
+ */
+public class ColonyViewWorkOrderMessage implements IMessage, IMessageHandler<ColonyViewWorkOrderMessage, IMessage>
+{
+    private int     colonyId;
+    private int     workOrderId;
+    private int     order;
+    private ByteBuf workOrderBuffer;
+    private workOrderMessages operation;
+
+    public enum workOrderMessages
+    {
+        EDIT,
+        EMPTY
+    }
+
+    /**
+     * Empty public constructor
+    */
+    public ColonyViewWorkOrderMessage()
+    {
+        /**
+         * Intentionally left empty.
+         **/
+    }
+
+    /**
+     * Updates a {@link com.minecolonies.colony.WorkOrderView} of the workOrders
+     *
+     * @param colony  Colony of the citizen
+     * @param workOrder Workorder of the colony to update view
+     */
+    public ColonyViewWorkOrderMessage(Colony colony, AbstractWorkOrder workOrder, int order, workOrderMessages operation)
+    {
+        this.colonyId = colony.getID();
+        this.workOrderBuffer = Unpooled.buffer();
+        this.order = order;
+        this.operation = operation;
+
+        if(workOrder != null)
+        {
+            this.workOrderId = workOrder.getID();
+            workOrder.serializeViewNetworkData(workOrderBuffer);
+        }
+    }
+
+    @Override
+    public void toBytes(ByteBuf buf)
+    {
+        buf.writeInt(colonyId);
+        buf.writeInt(workOrderId);
+        buf.writeInt(order);
+        buf.writeInt(operation.ordinal());
+        buf.writeBytes(workOrderBuffer);
+    }
+
+    @Override
+    public void fromBytes(ByteBuf buf)
+    {
+        colonyId = buf.readInt();
+        workOrderId = buf.readInt();
+        order = buf.readInt();
+        int op = buf.readInt();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'op' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)